.github/workflows/package.yml

```diff
@@ -0,0 +1,288 @@
+
+      - name: Prepare dist directory
+        run: |
+          mkdir -p dist_combined
+          # Copy service wheels first (only cp311 wheels)
+          cp dist/dist-ubuntu-24.04-x86_64-memory/*cp311*.whl dist_combined/
+          cp dist/dist-ubuntu-24.04-x86_64-s3/*cp311*.whl dist_combined/
+          # Copy main package
+          cp dist/dist-main/*.whl dist_combined/
+          cp dist/dist-main/*.tar.gz dist_combined/
+          echo "=== Contents of dist_combined ==="
+          ls -la dist_combined/
+
+      - name: Install packages in correct order
+        run: |
+          python -m pip install --upgrade pip
+          echo "=== Installing service packages ==="
+          # Install memory service (only cp311 wheels)
+          for whl in dist_combined/opendalfs_service_memory*cp311*.whl; do
+            if [ -f "$whl" ]; then
+              echo "Installing memory service: $whl"
+              pip install "$whl"
+            fi
+          done
+          # Install s3 service (only cp311 wheels)
+          for whl in dist_combined/opendalfs_service_s3*cp311*.whl; do
+            if [ -f "$whl" ]; then
+              echo "Installing s3 service: $whl"
+              pip install "$whl"
+            fi
+          done
+          echo "=== Installing main package ==="
+          pip install dist_combined/opendalfs-*.whl
+
+      - name: Verify imports
+        run: |
+          echo "=== Installed packages ==="
+          pip list | grep opendalfs
+          echo "=== Verifying imports ==="
+          python -c "import opendalfs"
+          python -c "import opendalfs_service_memory"
+          python -c "import opendalfs_service_s3"
+
+      - name: Test full installation
+        run: |
+          # Clean previous installation
+          pip uninstall -y opendalfs opendalfs_service_memory opendalfs_service_s3
+          # Install with all extras (use exact filename)
+          echo "=== Installing with all extras ==="
+          # First install service packages
+          for whl in dist_combined/opendalfs_service_*manylinux_2_17*.whl; do
+            if [ -f "$whl" ]; then
+              echo "Installing service: $whl"
+              pip install "$whl"
+            fi
+          done
+          # Then install main package with all extras
+          MAIN_WHEEL=$(find dist_combined -name "opendalfs-*.whl")
+          echo "Installing main package with extras: $MAIN_WHEEL"
+          pip install "$MAIN_WHEEL[all]"
+          # Verify imports
+          echo "=== Verifying imports ==="
+          python -c "import opendalfs"
+          python -c "import opendalfs_service_memory"
+          python -c "import opendalfs_service_s3"
+
+      - name: Cleanup
+        if: always()
+        run: |
+          pip uninstall -y opendalfs opendalfs_service_memory opendalfs_service_s3 || true
+          rm -rf dist dist_combined
+
+  # PyPI publishing job (commented out for future use)
```

Comment:
Let's use github OIDC for package uploads (so we don't need static tokens): https://github.com/apache/opendal/blob/6ca3eabc0cc5ad2b5e123acc215b62329bbdbf18/.github/workflows/release_python.yml#L105-L130

We can also upload to https://test.pypi.org/legacy/ to test and verify.

But I believe they are good to have in the future and not a blocker of merging this PR.

Reply:
Tracked at https://github.com/fsspec/opendalfs/issues/21